src/utils/jsx.js

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import * as t from "babel-types";
+import type {
+  BabelNodeExpression,
+  BabelNodeArrayExpression,
+  BabelNodeJSXElement,
+  BabelNodeJSXMemberExpression,
+  BabelNodeJSXIdentifier,
+  BabelNodeIdentifier,
+  BabelNodeMemberExpression,
+} from "babel-types";
+import invariant from "../invariant.js";
+
+const usedKeys = new Set();
```

Comment:
Don't maintain global state. Maybe wrap all of this code in a class, and have the serializer instantiate it for the duration of a serialization.